app/code/Magento/InventoryConfigurableProduct/Ui/DataProvider/Product/Form/Modifier/InventoryConfigurablePanel.php

```diff
@@ -52,10 +119,42 @@ private function getQuantityContainerConfig(): array
      */
     public function modifyMeta(array $meta)
     {
-        $matrix = $meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children'][ConfigurablePanel::CONFIGURABLE_MATRIX];
-        $matrix['children']['record']['children']['quantity_container'] = $this->getQuantityContainerConfig();
-        $meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children'][ConfigurablePanel::CONFIGURABLE_MATRIX] = $matrix;
+        if ($this->isSingleSourceMode->execute() === false) {
+            $meta[ConfigurablePanel::GROUP_CONFIGURABLE]['children']
```

Comment:
We can use `\Magento\Framework\Stdlib\ArrayManager`
Example in `\Magento\InventoryCatalog\Ui\DataProvider\Product\Form\Modifier\Quantity`

Reply:
Seems like it will not be useful here, because we introduce new field.